test/unit/puppetlabs/services/jruby/jruby_testutils.clj

```diff
@@ -29,6 +28,26 @@
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;; JRubyPuppet Test util functions
 
+(schema/defn ^:always-validate
+create-mock-pool-instance :- JRubyInstance
```

Comment:
During debugging this, we also thought there must be some tests not cleaning up after themselves or dirtying things up in some other way, so we did a little binary search, combining different tests with the failing one. After getting it down to 6 or 7 test namespaces, removing any of them would cause it to stop failing. Even then, it wouldn't fail every time. That makes me suspect it wasn't some dirty side effects from other tests specifically, but just from the amount of them